src/python/adafruit_BNO055/jaiabot_imu.py

```diff
@@ -85,71 +130,68 @@ def getData(self):
     imu = IMU()
 
 
-# Create socket
-port = args.port
-
-sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-sock.bind(('', port))
-
-imu_timeout = 5
-previous_time = time.time()
-
-while True:
-    data, addr = sock.recvfrom(1024) # buffer size is 1024 bytes
-
-    current_time = time.time()
-
-    if (previous_time + imu_timeout) <= current_time:
-        previous_time = current_time
-        print("reset imu")
-        imu.reset()
-
-    # Respond to anyone who sends us a packet
-    try:
-        data = imu.getData()
-    except Exception as e:
-        log.error(e)
-        continue
-    
-    now = datetime.utcnow()
-    euler = data['euler']
-    linear_acceleration = data['linear_acceleration']
-    gravity = data['gravity']
-    calibration_status = data['calibration_status'] # 1 is calibrated, 0 is not
-    try:
-        heading = euler[0]
-        print("=========================================")
-        print("")
-        formatted_heading_value_before = "Heading Before: {:.2f}".format(heading)
-        # Heading before
-        print(formatted_heading_value_before)
+def do_port_loop():
+    # Create socket
+    port = args.port
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.bind(('', port))
+
+    imu_timeout = 5
+    previous_time = time.time()
+
+    while True:
+
+        data, addr = sock.recvfrom(1024) # buffer size is 1024 bytes
+
+        current_time = time.time()
+
+        if (previous_time + imu_timeout) <= current_time:
+            previous_time = current_time
+            print("reset imu")
+            imu.reset()
```

Comment:
Do we still want to reset?
The reason I ask is if think quarernions fix our imu issues it would be good to comment this out for now to test.

Reply:
Good point - I commented it out.